frontend/public/components/debug-terminal.tsx

```diff
@@ -0,0 +1,235 @@
+import * as React from 'react';
+import { Alert } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import {
+  Firehose,
+  FirehoseResource,
+  FirehoseResult,
+  LoadingBox,
+  HorizontalNav,
+  PageHeading,
+} from '@console/internal/components/utils';
+import { PodKind, k8sCreate, k8sKillByName } from '@console/internal/module/k8s';
+import { PodExecLoader } from '@console/internal/components/pod';
+import { PodModel } from '@console/internal/models';
+
+import { resourcePath } from './utils/resource-link';
+import { getBreadcrumbPath } from '@console/internal/components/utils/breadcrumbs';
+
+const getDebugPod = async (
+  debugName: string,
+  podToDebug: PodKind,
+  containerName: string,
+): Promise<PodKind> => {
+  const container = podToDebug?.spec?.containers?.find((c) => c.name === containerName);
+  const isWindows = podToDebug?.spec?.tolerations?.some(
+    (t) => t.key === 'os' && t.value === 'Windows',
+  );
+  return {
+    kind: 'Pod',
```

Comment:
We should make a deep copy of the pod and change the values we need to. This won't work for a lot of pods because volume mounts, env, etc won't be there.

Reply:
done